docs/DeveloperGuide.md

```diff
@@ -1,38 +1,726 @@
+| v1.0    | Computing student who prefers typing over clicking | I can use keyboard instead of mouse                         | I can use the app efficiently                      |
+| v1.0    | Computing student who is too used to the Internet  | Reduce my browsing and information overload                 | I can improve my attention span                    |
+| v1.0    | Reflective student                                 | I can get one introspective question on-demand              | I can reflect and grow emotionally at my own pace  |
+| v1.0    | Computing student who wishes to improve lifestyle  | I can add an atomic habit to track                          | I can start the process of inculcating a new habit |
+| v1.0    | Computing student who wishes to improve lifestyle  | I can view all my atomic habits                             | I can keep track of my self-improvement progress   |
+| v1.0    | Computing student who wishes to improve lifestyle  | I can update my atomic habits                               | I can adjust the habits based on my progress       |
+| v2.0    | Reflective student                                 | I can mark introspective questions that I resonate with     | I can reflect using my favourite questions         |
+| v2.0    | Reflective student                                 | I can get the previous questions I viewed                   | I can re-view these questions                      |
+| v2.0    | Easily distracted computing student                | I want to start a timer to keep track of time spent on work | I can do timed-practice                            |
+| v2.0    | Easily distracted computing student                | I want to check the time                                    | I can keep track of my pace                        |
+| v2.0    | A regular WellNUS++ user                           | I wish to have my information stored in the app             | I can re-view my past data                         |
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+2. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should
+   be able to accomplish most of the tasks faster using commands than using the mouse.
 
 ## Glossary
 
 * *glossary item* - Definition
+* **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Main Command**: The first WORD that a user types in. `e.g. reflect, exit`
+* **Argument**: A word that is a parameter to a `Main Command` and is prefixed by ` --`. `e.g. --id, --name`
+* **Payload**: An (optional) arbitrary sequence of characters immediately following a main command or argument.
+  The payload will terminate when the user clicks `enter` or separates the payload with another argument
+  with the `--` delimiter.
 
 ## Instructions for manual testing
```

Comment:
Maybe this section could be removed from the developer guide? This is because your user guide already covers the commands, creating duplicate content.